src/main/java/com/google/step/servlets/PromoCodeServlet.java

```diff
@@ -1,19 +1,50 @@
 package com.google.step.servlets;
 
+import com.google.step.youtube.YoutubeInfoScraper;
+import com.google.step.youtube.PromoCode;
+import com.google.gson.Gson;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
 
 @WebServlet("/promo-code")
 public class PromoCodeServlet extends HttpServlet {
+
+    private YoutubeInfoScraper infoScraper;
+
+    public PromoCodeServlet(YoutubeInfoScraper infoScraper) {
+        this.infoScraper = infoScraper;
+    }
     
     @Override
     public void init() {
+        infoScraper = new YoutubeInfoScraper();
     }
 
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String channelId = request.getParameter("formInput");
+        try {
+            Optional<String> playlistId = infoScraper.scrapeChannelUploadPlaylist(channelId);
+            if (!playlistId.isPresent()) {
+                response.setContentType("application/json");
+                String json = new Gson().toJson(false);
+                response.getWriter().println(json);
```

Comment:
If you want, you can inline this so it's: response.getWriter().println(new Gson().toJson(false))